app/controllers/ems_common.rb

```diff
@@ -523,26 +524,27 @@ def retrieve_hawkular_security_protocols
      [_('Non-SSL'), 'non-ssl']]
   end
 
+  def add_success_flash_and_audit(emss, task)
+    add_flash(n_("%{task} initiated for %{count} %{model} from the %{product} Database",
+                 "%{task} initiated for %{count} %{models} from the %{product} Database", emss.length) % \
+      {:task    => task_name(task).gsub("Ems", ui_lookup(:tables => table_name)),
```

Comment:
I know you just moved this line, but it's too weird/complex :-(
`task_name` is defined in https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/helpers/application_helper/tasks.rb, has FIXME to be removed.
- `task_name("refresh_ems") == _("Refresh Provider")`.   gsub isn't used.
- `task_name("capture_ems")` will fall back to `titleize`, returns `"Capture Ems", gsub will turn it into "Capture Providers" [sic][*].

Perhaps you can just take an already human-ready `action_name` param instead of `task`, like in some other places here?

Also, the message already includes %{model}, no point saying it twice.  Currently is says:
`Capture Providers initiated for 2 Providers from the ManageIQ Database`
Could just say `Capture initiated for 2 Providers` or better `Metrics Capture initiated for 2 Providers`.

(IMHO we can also drop "from the ManageIQ Database" part?  I'd guess it originated from the "Delete initiated ... from the ManageIQ Database", where it actually adds information — that you're destroying the actual external system, just deleting it from our DB — and got copy-pasted to other actions where the "from" doesn't even make linguistic sense...)

[*] `ui_lookup(:table ...)` should return singular but for me it returns plural, same as `ui_lookup(:tables ...)`.  bug?

Reply:
👍 Didn't find an imoji for Oy Veyy :-)